frontend/src/components/Utils/filter-input.js

```diff
@@ -5,10 +5,40 @@ import { getUserUidArrFromUserEmailArr } from './temp-auth-utils.js'
 import { getTimestampFromDateString } from './time.js'
 import * as DB from '../../constants/database.js';
 
+/**
+ * An object containing the raw input data from a SaveTripModal form.
+ * @typedef {Object} RawTripData
+ * @property {string} title The trips's title.
+ * @property {string} description A description of the trip.
+ * @property {string} destination The general destination of the trip.
+ * @property {string} start_date Start date string in the form 'YYYY-MM-DD'.
+ * @property {string} end_date End date string in the form 'YYYY-MM-DD'.
+ * @property {!string[]} collaborators An array of collaborator emails.
+ */
+
+/**
+ * A trip object containing the data stored in a trip document in Firestore.
+ * @typedef {Object} TripData
+ * @property {string} title The trips's title.
+ * @property {string} description A description of the trip.
+ * @property {string} destination The general destination of the trip.
+ * @property {firebase.firestore.Timestamp} start_date Start date Firestore
+ *     timestamp object.
+ * @property {firebase.firestore.Timestamp} end_date End date Firestore
+ *     timestamp object
+ * @property {!string[]} collaborators An array of collaborator uids.
+ */
+
 /**
  * Return a string containing the cleaned text input.
  *
+ * For text element inputs, React has built in protections against injection/XSS
+ * attacks. Thus, no sanitization is needed for text inputs besides providing a
+ * default value in a trip field where applicable.
+ *
  * @param {string} rawInput String containing raw form input.
+ * @param {string} defaultValue The default value of the text input in case the
```

Comment:
I am not sure what you mean. I do not believe raw input will ever be null given that these are coming from text fields (they are always string). But I believe I could change the comparison operator to `==` instead of `===` to include null. I don't think there needs to be a change here but please follow up if I didn't interpret this right.